src/index.mjs

```diff
@@ -0,0 +1,49 @@
+import { downloadFromHub } from './controllers/downloadFiles.mjs';
+import { buildDataStructure } from './utils/hgDataStructure.mjs';
+import { fetchPageAndExtractURLs } from './utils/extractUrlsFromPage.mjs';
+import { log } from './utils/logger.mjs';
+import fastify from 'fastify'
+
+const server = fastify()
+
+server.get('/', async (request, reply) => {
+  try {
+    const huggingFaceURLs = await fetchPageAndExtractURLs();
+    const dataStructure = buildDataStructure(huggingFaceURLs);
+    log('info', 'Starting model downloads');
+
+    const downloadPromises = dataStructure.models.map(model => {
```

Comment:
Can we put some concurrency control here to get better performance?     Like limited max_concurrent_downloads to    a number, say 3 or 10     ....  perhaps in a dotev?

Reply:
Done!! 🚀